### DISCUSSION TITLE
How to implement optional locale routing?

### DISCUSSION THREAD

**Discussion by Flanker72** [nuxt/nuxt#17256] [Questions] [answered]
Need to implement the localized routing with optional locale token persisted in URL.
The main question is how to implement the default approach?
I need this structure:
```
/ - homepage for the predefined default locale
/catalog/[productId] - product page for the default locale
/[...slug] - any other page by slug for the default locale
/de/ - homepage for the German locale
/de/catalog/[productId] - product page for the German locale
/de/[...slug] - any other page for the German locale
```

As for now I cannot understand how to structure files within the `pages` folder.
Can you suggest any elegant solution for this?

**Comment by danielroe** [comment, marked as answer]
You could use the `pages:extend` hook to add more advanced route patterns to the paths of your pages.

Or try https://github.com/intlify/nuxt3 or the edge release of https://github.com/nuxt-community/i18n-module.
